### PR TITLE
LibWeb: Begin implementing infrastructure from the Fetch spec

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Statuses.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+{
+    // An ok status is a status in the range 200 to 299, inclusive.
+    return status >= 200 && status <= 299;
+}
+
+// https://fetch.spec.whatwg.org/#redirect-status
+bool is_redirect_status(Status status)
+{
+    // A redirect status is a status that is 301, 302, 303, 307, or 308.
+    return any_of(Array<u16, 5> { 301, 302, 303, 307, 308 }, [&](auto redirect_status) {
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Statuses.cpp:33]: Same here

File: Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Headers.cpp [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+};
+
+// https://fetch.spec.whatwg.org/#convert-header-names-to-a-sorted-lowercase-set
+ErrorOr<OrderedHashTable<ByteBuffer>> convert_header_names_to_a_sorted_lowercase_set(Span<ReadonlyBytes> header_names)
+{
+    // To convert header names to a sorted-lowercase set, given a list of names headerNames, run these steps:
+
+    // 1. Let headerNamesSet be a new ordered set.
+    Vector<ByteBuffer> header_names_set;
+    HashTable<ReadonlyBytes, CaseInsensitiveBytesTraits<u8 const>> header_names_seen;
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Headers.cpp:224]: Now does, good catch!

File: Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Headers.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+// https://fetch.spec.whatwg.org/#concept-header
+// A header is a tuple that consists of a name (a header name) and value (a header value).
+struct Header {
+    ByteBuffer name;
+    ByteBuffer value;
+};
+
+// https://fetch.spec.whatwg.org/#concept-header-list
+// A header list is a list of zero or more headers. It is initially the empty list.
+class HeaderList final : public Vector<Header> {
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Headers.h]: Added most of them now!

**Comment by linusg** [comment]
Thanks for the review! Fixed up all the issues, and rebased on top of the StringView changes :+1:

**Review by trflynn89 — APPROVED** [review]
LGTM! FYI it looks like the CI error is a clang bug:
https://github.com/llvm/llvm-project/issues/48582

**Pull request merged**